cpu/stm32f4/periph/uart.c

```diff
@@ -128,13 +130,15 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
         }
     }
     else {
+        mutex_lock(&_tx_lock[uart]);
         DMA_Stream_TypeDef *stream = dma_stream(uart_config[uart].dma_stream);
         /* configure and start DMA transfer */
         stream->M0AR = (uint32_t)data;
         stream->NDTR = (uint16_t)len;
         stream->CR |= DMA_SxCR_EN;
         /* wait for transfer to complete */
-        mutex_lock(&tx_sync[uart]);
+        mutex_lock(&_tx_dma_sync[uart]);
```

Comment:
@DipSwitch yes, the mutex is initially locked by `uart_init`. I merely changed the name of the mutex variable to better suit the purpose.